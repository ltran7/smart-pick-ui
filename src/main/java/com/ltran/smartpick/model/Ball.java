package com.ltran.smartpick.model;

import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor
public class Ball {

    private int number;
    private BallType type;

}
