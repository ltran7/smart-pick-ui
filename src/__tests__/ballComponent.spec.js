import React from "react";
import Ball from '../components/BallComponent';
import { create } from "react-test-renderer";

describe("Ball component", () => {

  test("Selected ball with id-1 and type-number", () => {
    const component = create(<Ball value={1} isSelected={true} onBallSelected={jest.fn()} type={"number"}/>);
    expect(component.toJSON()).toMatchSnapshot();

    const ball = component.root.findAllByType('button').find((b) => b.props.id.includes('1-number') && b.props.className.includes('ball-selected'));
    expect(ball).toBeDefined();
  });

  test("Selected ball with id-1 and type-star", () => {
    const component = create(<Ball value={1} isSelected={true} onBallSelected={jest.fn()} type={"star"}/>);
    expect(component.toJSON()).toMatchSnapshot();

    const ball = component.root.findAllByType('button').find((b) => b.props.id.includes('1-star') && b.props.className.includes('ball-selected'));
    expect(ball).toBeDefined();
  });

  test("Non selected ball with id-1 and type-number", () => {
    const component = create(<Ball value={1} isSelected={true} onBallSelected={jest.fn()} type={"number"}/>);
    expect(component.toJSON()).toMatchSnapshot();

    const ball = component.root.findAllByType('button').find((b) => b.props.id.includes('1-number') && b.props.className.includes('ball'));
    expect(ball).toBeDefined();
  });

  test("Non selected ball with id-1 and type-star", () => {
    const component = create(<Ball value={1} isSelected={true} onBallSelected={jest.fn()} type={"star"}/>);
    expect(component.toJSON()).toMatchSnapshot();

    const ball = component.root.findAllByType('button').find((b) => b.props.id.includes('1-star') && b.props.className.includes('ball'));
    expect(ball).toBeDefined();
  });

});