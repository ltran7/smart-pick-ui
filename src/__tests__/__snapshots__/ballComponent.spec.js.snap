// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ball component Non selected ball with id-1 and type-number 1`] = `
<button
  className="ball-selected"
  id="1-number"
  onClick={[Function]}
>
  1
</button>
`;

exports[`Ball component Non selected ball with id-1 and type-star 1`] = `
<button
  className="ball-selected"
  id="1-star"
  onClick={[Function]}
>
  1
</button>
`;

exports[`Ball component Selected ball with id-1 and type-number 1`] = `
<button
  className="ball-selected"
  id="1-number"
  onClick={[Function]}
>
  1
</button>
`;

exports[`Ball component Selected ball with id-1 and type-star 1`] = `
<button
  className="ball-selected"
  id="1-star"
  onClick={[Function]}
>
  1
</button>
`;
